sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.aest.controller.getApplicationOverallDetails.e2e-spec.ts

```diff
@@ -0,0 +1,127 @@
+import { HttpStatus, INestApplication } from "@nestjs/common";
+import * as request from "supertest";
+import {
+  AESTGroups,
+  BEARER_AUTH_TYPE,
+  createTestingAppModule,
+  getAESTToken,
+} from "../../../testHelpers";
+import {
+  createE2EDataSources,
+  E2EDataSources,
+  saveFakeApplication,
+} from "@sims/test-utils";
+import { Application, ApplicationStatus } from "@sims/sims-db";
+import { addDays } from "@sims/utilities";
+
+describe("ApplicationAESTController(e2e)-getApplicationOverallDetails", () => {
+  let app: INestApplication;
+  let db: E2EDataSources;
+
+  beforeAll(async () => {
+    const { nestApplication, dataSource } = await createTestingAppModule();
+    app = nestApplication;
+    db = createE2EDataSources(dataSource);
+  });
+
+  it("Should get all application versions in overall details when there is one or more edited application associated with the given application.", async () => {
+    // Arrange
+    // Create the parent application and also the first application version.
+    const firstVersionApplication = await saveFakeApplication(
+      db.dataSource,
+      undefined,
+      {
+        applicationStatus: ApplicationStatus.Edited,
+        submittedDate: addDays(-2),
+      },
+    );
+
+    const secondVersionApplication = await saveFakeApplication(
+      db.dataSource,
+      {
+        parentApplication: { id: firstVersionApplication.id } as Application,
+        precedingApplication: { id: firstVersionApplication.id } as Application,
+      },
+      {
+        applicationStatus: ApplicationStatus.Edited,
+        applicationNumber: firstVersionApplication.applicationNumber,
+        offeringIntensity:
+          firstVersionApplication.currentAssessment.offering.offeringIntensity,
+        submittedDate: addDays(-1),
+      },
+    );
+
+    const currentApplication = await saveFakeApplication(
+      db.dataSource,
+      {
+        parentApplication: { id: firstVersionApplication.id } as Application,
+        precedingApplication: { id: firstVersionApplication.id } as Application,
```

Comment:
It will not change the test output but should it not be a different `precedingApplication` for this one?